nsqd/stats.go

```diff
@@ -113,20 +113,30 @@ type ChannelsByName struct {
 
 func (c ChannelsByName) Less(i, j int) bool { return c.Channels[i].name < c.Channels[j].name }
 
-func (n *NSQD) GetStats() []TopicStats {
+func (n *NSQD) GetStats(topic string, channel string) []TopicStats {
 	n.RLock()
-	realTopics := make([]*Topic, 0, len(n.topicMap))
-	for _, t := range n.topicMap {
-		realTopics = append(realTopics, t)
+	var realTopics []*Topic
+	if topic == "" {
+		realTopics = make([]*Topic, 0, len(n.topicMap))
+		for _, t := range n.topicMap {
+			realTopics = append(realTopics, t)
+		}
+	} else {
+		realTopics = []*Topic{n.topicMap[topic]}
```

Comment:
Just realized we're not checking if the topic exists in `topicMap`, can you add that check here?

Reply:
@mreiferson Good catch! I also have enhanced the test for both of none exist topic and channel.

PTAL.